### PR TITLE
REP-5727 Logging Trace ID Log Messages With New Logger

### PR DESCRIPTION
**Pull request by dmnjohns**

### CODE REVIEW

File: repose-aggregator/src/docs/asciidoc/release-notes.adoc [hunk 1]
```diff
@@ -13,6 +13,16 @@
...
+* https://repose.atlassian.net/browse/REP-5727[REP-5727] - Extracted trace ID logging to its own named logger.
+** The `trace-id-logging` Logger in your Log4j2 configuration will determine the logging behavior for trace ID logging.
+If the `trace-id-logging` Logger has not been configured, it will inherit the `Root` logger's configuration.
++
+[IMPORTANT]
+====
+If Repose is already installed and configured, and the `Root` logger is not configured to log at the `trace` logging level, then upgrading Repose will drop some or all trace ID logging messages.
+Please add and configure the `trace-id-logging` Logger to regain those messages.
+====
+
```
Comment (dmnjohns) [inline repose-aggregator/src/docs/asciidoc/release-notes.adoc]: So the naive approach could change logging behavior underneath people that upgrade, but already have a `log4j2` configuration in place. It is somewhat likely that they will lose at least the trace level trace ID log messages. Does that have some implication with regards to the version of the release this work can be in?

File: repose-aggregator/src/docs/asciidoc/release-notes.adoc [hunk 2]
```diff
@@ -13,6 +13,16 @@
 *** https://github.com/rackerlabs/api-checker/blob/api-checker-2.4.0/RELEASE.md[API Checker v2.4.0 release notes]
 ** Attribute Mapper: 1.3.0 → 1.3.1
 *** https://github.com/rackerlabs/attributeMapping/blob/attribute-mapper-1.3.1/RELEASE.md[Attribute Mapper v1.3.1 release notes]
+* https://repose.atlassian.net/browse/REP-5727[REP-5727] - Extracted trace ID logging to its own named logger.
+** The `trace-id-logging` Logger in your Log4j2 configuration will determine the logging behavior for trace ID logging.
+If the `trace-id-logging` Logger has not been configured, it will inherit the `Root` logger's configuration.
++
+[IMPORTANT]
```
Comment (dmnjohns) [inline repose-aggregator/src/docs/asciidoc/release-notes.adoc]: Important? Warning?

**Comment by adrianjgeorge** [comment]
So it just occurred to me that we can achieve what we want without breaking compatibility in anyway. Instead of making this simply named logger, make it a sub logger of the of power filter.
`org.openrepose.powerfilter.PowerFilter.trace-id-logging`
Anyone who has tweaked logging of repose classes will keep their current behavior, but we still add the granularity to turn this off.

**Comment by dmnjohns** [comment]
Man, that is a great idea that totally did not occur to me. Will do.

File: repose-aggregator/artifacts/src/config/filters/log4j2.xml [hunk 3]
```diff
@@ -30,6 +30,7 @@
         <Logger name="org.springframework" level="warn"/>
         <Logger name="intrafilter-logging" level="info"/>
         <Logger name="org.openrepose.commons.utils.servlet.http.HttpServletResponseWrapper.headerWarning" level="warn"/>
+        <Logger name="trace-id-logging" level="trace"/>
```
Comment (dmnjohns) [inline repose-aggregator/artifacts/src/config/filters/log4j2.xml]: Looking at this again, should I set this to `info`? That is the current default for the `PowerFilter` logger, so I'm thinking probably so.

File: repose-aggregator/src/docs/asciidoc/release-notes.adoc (hunk 2 above)
Comment (dmnjohns) [reply repose-aggregator/src/docs/asciidoc/release-notes.adoc]: This gets dropped with the child logger change.

**Comment by adrianjgeorge** [comment]
rebase so the jenkins build works

**Comment by dmnjohns** [comment]
Rebased.

**Pull request merged**